dace/transformation/dataflow/vectorization.py

```diff
@@ -18,100 +143,341 @@ class Vectorization(transformation.SingleStateTransformation):
+
+                    edge_subset = [a_tuple[0] for a_tuple in list(e.data.subset)]
+
+                    if isinstance(edge_subset[-1], symbol) and str(edge_subset[-1]) != map_subset[-1]:
+                        return False
+
+            # Not possible to handle interstate edges
+
+            for e, _ in sdfg.all_edges_recursive():
+
+                if isinstance(e.data, InterstateEdge):
+
+                    if e.data.assignments != {} or e.data.condition.as_string != "1":
+                        return False
+
+            self.map_entry = old_map_entry
+            self._level = 0
+
+            # Run the vector inference algorithm to check if vectorization is feasible
+        try:
+            vector_inference.infer_vectors(sdfg,
+                                           state,
+                                           map_entry,
+                                           self.vector_len,
+                                           flags=vector_inference.VectorInferenceFlags.Allow_Stride,
+                                           strided_map=self.strided_map
+                                           or self.target == dtypes.ScheduleType.FPGA_Device,
+                                           apply=False)
+        except vector_inference.VectorInferenceException as ex:
+            return False
+
+        return True
 
     def apply(self, graph: SDFGState, sdfg: SDFG):
         map_entry = self.map_entry
         tasklet: nodes.Tasklet = graph.successors(map_entry)[0]
         param = symbolic.pystr_to_symbolic(map_entry.map.params[-1])
 
-        # Create new vector size.
+        # To support recursivity in the FPGA case, see below
+        if self._level == 0:
+            state = graph
+        else:
+            state = get_innermost_state_for_node(sdfg, map_entry)
+
+        if self.target == dtypes.ScheduleType.FPGA_Device and self._level == 0:
+
+            maps_to_vectorize, data_descriptors_to_vectorize = collect_maps_to_vectorize(sdfg, state, map_entry)
+
+            old_map_entry = self.map_entry
+            self._level = 1  # To prevent infinte loop
+
+            for m in maps_to_vectorize:
+                self.map_entry = m
+
+                self.apply(get_innermost_state_for_node(sdfg, m), get_innermost_sdfg_for_node(sdfg, m))
+
+            # Change the subset in the post state that copies the data back to the host
+            if len(sdfg.states()) < 3 or not sdfg.states()[-1].name.startswith('post_'):
+                # FPGA Transformation not yet applied
+                return
+
+            for e in sdfg.states()[-1].edges():
+
+                if e.data.data in data_descriptors_to_vectorize:
+
+                    desc = sdfg.arrays[e.data.data]
+                    contigidx = desc.strides.index(1)
+                    lastindex = e.data.subset[contigidx]
+                    i, j, k = lastindex
+                    e.data.subset[contigidx] = (i, (j + 1) / self.vector_len - 1, k)
```

Comment:
Does this refer to `start`, `stop`, and `step`? Don't use `i`, `j`, `k`, it looks like something is hardcoded to 3D indices